src/axom/sina/core/AdiakWriter.hpp

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) 2017-2024, Lawrence Livermore National Security, LLC and
+// other Axom Project Developers. See the top-level LICENSE file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+#ifndef SINA_ADIAK_HPP
+#define SINA_ADIAK_HPP
+
+/*!
+ ******************************************************************************
+ *
+ * \file AdiakWriter.hpp
+ *
+ * \brief   Header file for the Adiak Sina callback function.
+ *
+ ******************************************************************************
+ */
+
+#include "axom/config.hpp"
+#ifdef AXOM_USE_ADIAK
+
+  #include <string>
+  #include <type_traits>
+
+  #include "axom/sina/core/ConduitUtil.hpp"
+  #include "axom/sina/core/Record.hpp"
+  #include "axom/sina/core/Run.hpp"
+
+extern "C" {
+  #include "adiak_tool.h"
+}
+
+namespace axom
+{
+namespace sina
+{
+
+/**
+ * \brief The callback function to pass to Adiak in order to write collected data to a Sina Record.
+ *
+ * To register this with Adiak, you should call adiak_register_cb, passing in
+ * a pointer to the record as the last value. Your code should look something
+ * like this:
+ *
+ * \code
+ *     axom::sina::Record record{ID{"my_id", axom::sina::IDType::Local}, "my_record_type"};
+ *     axom::sina::adiak_register_cb(1, adiak_category_all, axom::sina::adiakSinaCallback, 0, &record);
```

Comment:
That did not work for me (might be missing some include or something? I had to do:
#include "adiak.hpp"
extern "C" {
#include "adiak_tool.h"
#include "adiak.h"
}
and then call with:
adiak_register_cb(1, adiak_category_all, axom::sina::adiakSinaCallback, 0, &record);

Reply:
You're right this needs updated. I just tried with:
```
#include "axom/sina.hpp"

int main(void)
{
  axom::sina::Record record{axom::sina::ID{"my_id", axom::sina::IDType::Local}, "my_record_type"};
  adiak_register_cb(1, adiak_category_all, axom::sina::adiakSinaCallback, 0, &record);
}
```
And that seemed to work. Can you let me know if that works for you as well?